pkg/plugins/localfiles.go

```diff
@@ -10,79 +11,189 @@ import (
 	"github.com/grafana/grafana/pkg/util"
 )
 
-var _ fs.FS = (*LocalFS)(nil)
+// CollectFilesFunc is a function that returns a filepath.WalkFunc, which will accumulate its results into acc.
+// The file paths collected into acc should be absolute file paths (like the first argument provided to
+// filepath.WalkFunc)
+type CollectFilesFunc func(acc map[string]struct{}) filepath.WalkFunc
+
+// defaultCollectFilesFunc is a simple CollectFilesFunc that skips all directories and collects all paths into acc.
+var defaultCollectFilesFunc CollectFilesFunc = func(acc map[string]struct{}) filepath.WalkFunc {
+	return func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+		acc[path] = struct{}{}
+		return nil
+	}
+}
+
+// emptyCollectFilesFunc is a no-op CollectFilesFunc. It returns a filepath.WalkFunc that does nothing and
+// the accumulator map is not altered in any way. This is suitable to use with plugins.allowListFSNoFiles for tests.
+func emptyCollectFilesFunc(_ map[string]struct{}) filepath.WalkFunc {
+	return func(path string, info fs.FileInfo, err error) error {
+		return nil
+	}
+}
```

Comment:
can you move this and `NewAllowListLocalFSForTests` to a different file? (to avoid mixing test and functional logic). No need to use a different package, a `test_utils.go` should work.